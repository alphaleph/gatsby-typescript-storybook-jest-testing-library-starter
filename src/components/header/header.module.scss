.container {
    background: rebeccapurple;
    margin-bottom: 1.45rem;
}

.dark {
    @extend .container;
    background: rgb(34, 5, 34);
}

.content {
    margin: 0 auto;
    max-width: 960;
    padding: 1.45rem 1.0875rem;
}

.mainHeader {
    margin: 0;
}

.link {
    color: white;
}