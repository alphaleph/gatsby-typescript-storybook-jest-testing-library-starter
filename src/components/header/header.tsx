import { Link } from 'gatsby';
import React from 'react';
import styles from './header.module.scss';

interface HeaderProps {
    siteTitle: string;
    isDarkMode?: boolean;
    test: number
}

export const Header = ({ isDarkMode = false, ...props }: HeaderProps) => {
    return (
        <header className={isDarkMode ? styles.dark : styles.container}>
            <div className={styles.content}>
                <h1 className={styles.mainHeader}>
                    <Link to="/" className={styles.link}>
                        {props.siteTitle}
                    </Link>
                </h1>
            </div>
        </header>
    );
};
