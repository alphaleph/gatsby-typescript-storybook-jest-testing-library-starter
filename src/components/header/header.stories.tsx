import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, object } from '@storybook/addon-knobs';
import { Header } from './header';

export const siteTitle = 'Gatsby-typescript-storybook-jest-testing-library Starter';
export const isDarkMode = false;
export const actions = {
    onDarkMode: action('onDarkMode'),
};

storiesOf('Header', module)
    .addDecorator(withKnobs)
    .add('default', () => (
        <Header
            siteTitle={object('siteTitle', siteTitle)}
            isDarkMode={object('isDarkMode', isDarkMode)}
            {...actions}
        />
    ))
    .add('dark-mode', () => (
        <Header siteTitle={siteTitle} isDarkMode={true} {...actions} />
    ));
