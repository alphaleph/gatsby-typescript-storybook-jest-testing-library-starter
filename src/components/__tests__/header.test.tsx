import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../header/header';
import '@testing-library/jest-dom/extend-expect';

const siteTitle = 'Gatsby-typescript-storybook-jest-testing-library Starter';

describe('Header', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<Header siteTitle={siteTitle} />);
        expect(getByText(siteTitle)).toBeInTheDocument();
    });
    it('renders correctly', () => {
        const { getByText } = render(<Header siteTitle={siteTitle} />);
        expect(getByText(siteTitle).textContent).toBe(siteTitle);
    });
    it('has not changed', () => {
        const { asFragment } = render(<Header siteTitle={siteTitle} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
