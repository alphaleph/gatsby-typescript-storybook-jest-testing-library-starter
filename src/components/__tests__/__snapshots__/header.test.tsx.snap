// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has not changed 1`] = `
<DocumentFragment>
  <header
    class="container"
  >
    <div
      class="content"
    >
      <h1
        class="mainHeader"
      >
        <a
          class="link"
          href="/"
        >
          Gatsby-typescript-storybook-jest-testing-library-starter
        </a>
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
